classes/doi/RegistrationAgencySettings.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * @file classes/doi/RegistrationAgencySettings.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class RegistrationAgencySettings
+ * @ingroup doi
+ *
+ * @brief Base class for registration agency plugin settings management
+ */
+
+namespace PKP\doi;
+
+use APP\core\Services;
+use APP\plugins\IDoiRegistrationAgency;
+use Illuminate\Validation\Validator;
+use PKP\components\forms\Field;
+use PKP\context\Context;
+use PKP\plugins\Hook;
+use PKP\services\interfaces\EntityWriteInterface;
+use PKP\services\PKPSchemaService;
+use PKP\validation\ValidatorFactory;
+
+abstract class RegistrationAgencySettings
```

Comment:
That makes sense. I like the idea of moving the updating back to `PKPContextHandler()` (it seems much more obvious looking at it now). Are there any other changes/different ways of approaching this you would recommend to avoid too much coupling?

Reply:
The main thing would be to split the form fields from the settings/validation. But I don't think it's a big priority.

Reply:
Sounds good. Happy to leave it as you've suggested for now. Was mostly curious for next time so I don't use this as a pattern to follow in the future.